sql/src/main/java/io/crate/execution/engine/sort/InputFieldComparator.java

```diff
@@ -88,7 +88,18 @@ public int compareBottom(int doc) throws IOException {
         for (LuceneCollectorExpression collectorExpression : collectorExpressions) {
             collectorExpression.setNextDocId(doc);
         }
-        return valueType.compareValueTo(bottom, MoreObjects.firstNonNull(input.value(), missingValue));
+        return valueType.compareValueTo(bottom, getFirstNonNullOrNull(input.value(), missingValue));
+    }
+
+    @Nullable
+    private Object getFirstNonNullOrNull(Object first, Object second) {
```

Comment:
could be static

Reply:
Just a suggestion you could do:
```
Optional.ofNullable(first).orElse(second)
```

Reply:
Please avoid the nullable instance creation. Granted, escape analysis will probably get rid of it again, but that's just necessary for a simple if check.